p-token/tests/withdraw_excess_lamports.rs

```diff
@@ -0,0 +1,756 @@
+    let context = ProgramTest::new("pinocchio_token_program", TOKEN_PROGRAM_ID, None)
+        .start_with_context()
+        .await;
+
+    let excess_lamports = 4_000_000_000_000;
+
+    // Given an account
+
+    let multisig = Keypair::new();
+    let signer1 = Keypair::new();
+    let signer1_pubkey = signer1.pubkey();
+    let signer2 = Keypair::new();
+    let signer2_pubkey = signer2.pubkey();
+    let signer3 = Keypair::new();
+    let signer3_pubkey = signer3.pubkey();
+    let signers = vec![&signer1_pubkey, &signer2_pubkey, &signer3_pubkey];
+
+    let rent = context.banks_client.get_rent().await.unwrap();
+    let account_size = size_of::<Multisig>();
+
+    let mut initialize_ix = spl_token::instruction::initialize_multisig(
+        &spl_token::ID,
+        &multisig.pubkey(),
+        &signers,
+        3,
+    )
+    .unwrap();
+    // Switches the program id to the token program.
+    initialize_ix.program_id = token_program;
+
+    // And we initialize the multisig account.
+
+    let instructions = vec![
+        system_instruction::create_account(
+            &context.payer.pubkey(),
+            &multisig.pubkey(),
+            rent.minimum_balance(account_size) + excess_lamports,
+            account_size as u64,
+            &token_program,
+        ),
+        initialize_ix,
+    ];
+
+    let tx = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &multisig],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    let account = context
+        .banks_client
+        .get_account(multisig.pubkey())
+        .await
+        .unwrap();
+
+    assert!(account.is_some());
+
+    let account = account.unwrap();
+    assert_eq!(
+        account.lamports,
+        rent.minimum_balance(account_size) + excess_lamports
+    );
+
+    // When we try to withdraw the excess lamports with the wrong authority.
+
+    let destination = Pubkey::new_unique();
+    let wrong_authority = Keypair::new();
+
+    let mut withdraw_ix = spl_token_2022::instruction::withdraw_excess_lamports(
+        &spl_token_2022::ID,
+        &multisig.pubkey(),
+        &destination,
+        &wrong_authority.pubkey(),
+        &signers,
+    )
+    .unwrap();
+    // Switches the program id to the token program.
+    withdraw_ix.program_id = token_program;
+
+    let tx = Transaction::new_signed_with_payer(
+        &[withdraw_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &signer1, &signer2, &signer3],
+        context.last_blockhash,
+    );
+    let error = context
+        .banks_client
+        .process_transaction(tx)
+        .await
+        .unwrap_err();
+
+    // The we expect an error.
+
+    assert_matches!(
+        error,
+        BanksClientError::TransactionError(TransactionError::InstructionError(
+            _,
+            InstructionError::Custom(4) // TokenError::OwnerMismatch
```

Comment:
You could potentially make this clearer by casting the error, ie 
```suggestion
            InstructionError::Custom(TokenError::OwnerMismatch as u32)
```

Reply:
Looks like the macro doesn't like that, so let's just merge this as is